datashuttle/utils_mod/utils.py

```diff
@@ -239,25 +240,132 @@ def process_names(
             "Ensure subject and session names are list of strings, or string"
         )
 
-    if any([" " in ele for ele in names]):
-        raise_error("sub or ses names cannot include spaces.")
-
     if isinstance(names, str):
         names = [names]
 
-    update_names_with_datetime(names)
+    if any([" " in ele for ele in names]):
+        raise_error("sub or ses names cannot include spaces.")
 
     prefixed_names = ensure_prefixes_on_list_of_names(names, prefix)
 
     if len(prefixed_names) != len(set(prefixed_names)):
         raise_error(
-            "Subject and session names but all be unqiue (i.e. there are no"
+            "Subject and session names but all be unique (i.e. there are no"
             " duplicates in list input)"
         )
 
+    prefixed_names = update_names_with_range_to_flag(prefixed_names, prefix)
+
+    update_names_with_datetime(prefixed_names)
+
     return prefixed_names
 
 
+# Handle @TO flags  -------------------------------------------------------
+
+
+def update_names_with_range_to_flag(names: list, prefix: str) -> list:
+    """
+    Given a list of names, check if they contain the @TO keyword.
+    If so, expand to a range of names. Names including the @TO
+    keyword must be in the form prefix-num1@num2. The maximum
+    number of leading zeros are used to pad the output
+    e.g.
+    sub-01@003 becomes ["sub-001", "sub-002", "sub-003"]
+
+    Input can also be a mixed list e.g.
+    names = ["sub-01", "sub-02@TO04", "sub-05@TO10"]
+    will output a list of ["sub-01", ..., "sub-10"]
+    """
+    new_names = []
+
+    for i, name in enumerate(names):
+
+        if "@TO" in name:
+
+            check_name_is_formatted_correctly(name, prefix)
+
+            prefix_tag = re.search(f"{prefix}[0-9]+@TO[0-9]+", name)[0]  # type: ignore
+            tag_number = prefix_tag.split(f"{prefix}")[1]
+
+            name_start_str, name_end_str = name.split(tag_number)
+
+            if "@TO" not in tag_number:
+                raise_error(
+                    f"@TO flag must be between two numbers in the {prefix} tag."
+                )
+
+            left_number, right_number = tag_number.split("@TO")
+
+            if int(left_number) >= int(right_number):
+                raise_error(
+                    "Number of the subject to the  left of @TO flag "
```

Comment:
Shouldn't this be "Number of {prefix}"? The error may be in session rather than in subject